### PR TITLE
[v12] Cleanup & Improvements

### PR DESCRIPTION
**Pull request by Reecepbcups**
**Cleanup & Improvements**

note: real PR is only like 700 lines, the 90k+ are auto generated

CI:
- Adds concurrency to cancel workflows for a PR on new commit
- CI lint `price-feeder`. Closes: #467 
- Fix oracle test to actually run in CI

Docs:
- removes screen from oracle docs

FeeShare:
- Fixed RPC queries from .../FeeShare/ to /fee_share/ for swagger

Price Feeder
- TEST: use JUNOUSDC pool instead of JUNORAW (bc migration soon)
- TEST: use proper binance websocket endpoint (used in config as well)
- Log which provider timeout from the feeder log (always osmosis & juno APIs)

Other:
- Variable timeout commit for docker compose (6s to 500ms per block) _(Faster CI & local testing)_
- CLI: [junod debug (forceprune | export-derive-balances | staked-to-csv)](https://github.com/CosmosContracts/juno/pull/509/commits/7f73af8f43c12e78e4ab8bb79e1d7f625afbdc3b) -> [proof](https://gist.github.com/Reecepbcups/411a35a9005ddcec4ecab1a2a3605fab)
- Update `add-ica-config` to new messages v12 c3c1a80d9237c84020ae6e584b3f375363dd565c

- Swagger now has globalfee, Tokenfactory, InterTx, PacketForwardMiddleware and CosmWasm endpoints shown

- Spelling

### CODE REVIEW

**Comment by faddat** [comment]
Personally, I'd keep screen.

It isn't the best way to run the oracle, but the difference is that it allows operators to see what is going on.

**Comment by Reecepbcups** [comment]
Is `add-ica-config` required to sync from older versions @the-frey or just local testing?

If so I will make c3c1a80d9237c84020ae6e584b3f375363dd565c the new `add-ica-config`, and move previous messages to `v11-add-ica-config`. Just wanted to double check

**Comment by the-frey** [comment]
the add ica config command is to help generating genesis files that match mainnet config. So yeah, docker, testing, localnets, stuff like that. So what you've done is right, we'd want v12 to have it, or main just after v12. What you've done makes sense tho

**Comment by Reecepbcups** [comment]
@udit-gulati So the CI was not correctly running for the oracle on main, which is why your other pr #514 passed. 

The test actually fails for grpc query "Error - Start time before first entry" as you can see above. 
Can you target a PR to this branch & resolve that? Thanks

**Comment by udit-gulati** [comment]
@Reecepbcups it now calculates TWAP when start time is before first entry, I've updated the test https://github.com/CosmosContracts/juno/pull/517. Let me know if this shouldn't be the expected behaviour.

**Comment by jackzampolin** [comment]
plz no screen

**Comment by the-frey** [comment]
LGTM

**Review by faddat — APPROVED** [review]
Approved with the note that force prune will probably break in the next version.

**Comment by faddat** [comment]
Oh, you fixed the thing I was fixing.  Strong approve.

**Comment by faddat** [comment]
--- FAIL: TestOracleTestSuite (0.04s)
    --- FAIL: TestOracleTestSuite/TestEndblockerHistoracle (0.04s)
        abci_test.go:155: 
            	Error Trace:	/home/runner/work/juno/juno/x/oracle/abci_test.go:155
            	Error:      	Should be true
            	Test:       	TestOracleTestSuite/TestEndblockerHistoracle
FAIL


I... need to fix spelling sir.  

ALso need to know if that is HistOracle

or Historical

**Review by faddat — CHANGES_REQUESTED** [review]
brb

**Comment by Reecepbcups** [comment]
> --- FAIL: TestOracleTestSuite (0.04s) --- FAIL: TestOracleTestSuite/TestEndblockerHistoracle (0.04s) abci_test.go:155: Error Trace: /home/runner/work/juno/juno/x/oracle/abci_test.go:155 Error: Should be true Test: TestOracleTestSuite/TestEndblockerHistoracle FAIL
> 
> I... need to fix spelling sir.
> 
> ALso need to know if that is HistOracle
> 
> or Historical

This randomly fails sometimes, then passes others. Its like e2e. IDK why oracle is so weird like this

**Comment by faddat** [comment]
Interesting about the failure -- my changes really just aim at readability

Historacle

sounds too. much like "historical"

**Review by faddat — APPROVED** [review]

**Comment by faddat** [comment]
...he says, while he makes a spelling error in a commit message...

File: cmd/junod/cmd/balances_from_state_export.go [hunk 1]
```diff
@@ -0,0 +1,269 @@
...
+
+			err = os.WriteFile(snapshotOutput, snapshotJSON, 0o600)
+			return err
+		},
+	}
+
+	return cmd
+}
+
+// StakedToCSVCmd generates a airdrop.csv from a provided exported balances.json.
```
Comment (JakeHartnell) [inline cmd/junod/cmd/balances_from_state_export.go:199]: OMG, this is soooooo nice.

File: cmd/junod/cmd/forceprune.go [hunk 2]
```diff
@@ -0,0 +1,213 @@
...
+	validators        = "validatorsKey:"
+	consensusParams   = "consensusParamsKey:"
+	ABCIResponses     = "abciResponsesKey:"
+	fullHeight        = "full_height"
+	minHeight         = "min_height"
+	defaultFullHeight = "188000"
+	defaultMinHeight  = "1000"
+)
+
+// forceprune gets cmd to convert any bech32 address to an osmo prefix.
```
Comment (JakeHartnell) [inline cmd/junod/cmd/forceprune.go:36]:
```suggestion
```

Comment is wrong. It does not do this. :joy:

**Review by JakeHartnell — APPROVED** [review]
BEAUTIFUL. :shipit:

**Pull request merged**